pkg/resource/stack/deployment.go

```diff
@@ -99,28 +100,34 @@ func DeserializeUntypedDeployment(deployment *apitype.UntypedDeployment) (*deplo
 		return nil, ErrDeploymentSchemaVersionTooOld
 	}
 
-	var v2deployment apitype.DeploymentV2
+	var v3deployment apitype.DeploymentV3
 	switch deployment.Version {
 	case 1:
 		var v1deployment apitype.DeploymentV1
 		if err := json.Unmarshal([]byte(deployment.Deployment), &v1deployment); err != nil {
 			return nil, err
 		}
-
-		v2deployment = migrate.UpToDeploymentV2(v1deployment)
+		v2deployment := migrate.UpToDeploymentV2(v1deployment)
+		v3deployment = migrate.UpToDeploymentV3(v2deployment)
 	case 2:
-		if err := json.Unmarshal([]byte(deployment.Deployment), &v2deployment); err != nil {
+		//var v2deployment apitype.DeploymentV2
```

Comment:
I'm assuming you're going to uncomment this when we stop lying to the service about the checkpoint version, but don't forget 😄